### PR TITLE
Improve sanitizer builds

### PR DESCRIPTION
**Pull request by mpoeter**
### Scope & Purpose

Enterprise companion https://github.com/arangodb/enterprise/pull/1453

This PR disables some long running/memory intensive tests for tsan and also improves some tests to use babies operations.

### CODE REVIEW

File: tests/js/client/aql/aql-index-cache.js [hunk 1]
```diff
@@ -1081,17 +1082,20 @@ function PersistentIndexUniqueReadOnlyStoredValuesSuite() {
...
-jsunity.run(PersistentIndexNonUniqueHugeValueSuite);
-jsunity.run(PersistentIndexNonUniqueReadOnlyCacheDisabledSuite);
-jsunity.run(PersistentIndexNonUniqueReadOnlyCacheEnabledSuite);
-jsunity.run(PersistentIndexNonUniqueReadOnlyStoredValuesSuite);
-jsunity.run(PersistentIndexUniqueReadOnlyCacheDisabledSuite);
-jsunity.run(PersistentIndexUniqueReadOnlyCacheEnabledSuite);
-jsunity.run(PersistentIndexUniqueReadOnlyStoredValuesSuite);
-jsunity.run(OtherIndexesSuite);
+// disable these tests for cluster sanitizer runs since they are very slow
+if (!isCluster || (!versionHas('asan') && !versionHas('tsan'))) {
```
Comment (dothebart) [inline tests/js/client/aql/aql-index-cache.js:1086]: No magic execution blocks inside testfiles!
Use filename extensions `-noasan` and `-noncluster`

**Review by dothebart — CHANGES_REQUESTED** [review]
please don't skip inside testfiles. 
Files should be ruled out by suffixes so its clearly visible they won't be executed under certain circumstances.

**Review by dothebart — APPROVED** [review]
LGTM.

File: tests/js/client/aql/aql-index-cache-noinstr_or_noncluster.js [hunk 2]
```diff
@@ -32,7 +32,10 @@ const internal = require('internal');
 const { deriveTestSuite } = require('@arangodb/test-helper');
 const isCluster = require("internal").isCluster();
 const canUseFailAt =  internal.debugCanUseFailAt();
+const versionHas = require("@arangodb/test-helper").versionHas;
```
Comment (mpoeter) [inline tests/js/client/aql/aql-index-cache-noinstr_or_noncluster.js:35]:
```suggestion
```

**Pull request merged**